blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubScm.java

```diff
@@ -230,7 +227,7 @@ public HttpResponse validateAndCreate(@JsonBody JSONObject request) {
             //Now we know the token is valid. Lets find credential
             StandardUsernamePasswordCredentials githubCredential = CredentialsUtils.findCredential(getId(), StandardUsernamePasswordCredentials.class, new BlueOceanDomainRequirement());
 
-            final StandardUsernamePasswordCredentials credential = new UsernamePasswordCredentialsImpl(CredentialsScope.USER, "github", "Github Access Token", authenticatedUser.getId(), accessToken);
+            final StandardUsernamePasswordCredentials credential = new UsernamePasswordCredentialsImpl(CredentialsScope.USER, getId(), "Github Access Token", authenticatedUser.getId(), accessToken);
```

Comment:
One credential per URL makes sense. But the UI needs to make an initial fetch to know which GitHub Enterprise credentials have been saved. Something like...

`/blue/rest/organizations/${orgname}/scm/github-enterprise`
returns
`[ { apiUrl: "https://ghe.mycorp1.com" }, { apiUrl: "https://ghe.somellc2.com" } ]`

And if multiple credentials do exist, the user needs a way to pick one. Unless you're thinking that each endpoint corresponds to an organization? I suppose that's possible but it seems like a mismatch to me.